### PR TITLE
MM-56980 add shared channel icon to searchable channel list

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
Checks and displays the shared channel icon in the searchable channel list.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-56980

#### Screenshots
Screenshot 2024-03-04 at 2 23 33 PM

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/searchable_channel_list.tsx [hunk 1]
```diff
@@ -127,7 +128,15 @@ export class SearchableChannelList extends React.PureComponent<Props, State> {
         const ariaLabel = `${channel.display_name}, ${channel.purpose}`.toLowerCase();
         let channelTypeIcon;
 
-        if (isArchivedChannel(channel)) {
+        if (channel.shared) {
```
Comment (calebroseland) [inline webapp/channels/src/components/searchable_channel_list.tsx:131]: Can a shared channel be archived? (Or does achieving a channel remove its shared status?)

Comment (sbishel) [reply webapp/channels/src/components/searchable_channel_list.tsx]: Yep...I moved this up. So that "archive" takes presedence.

**Review by calebroseland — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]
🎉

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - I'll test this post-merge as it's not easy to set up shared channels on the spinwick

**Pull request merged**